src/main/java/sirius/kernel/async/CallContext.java

```diff
@@ -353,10 +353,46 @@ public <C extends SubContext> C get(Class<C> contextType) {
      * @param instance    the instance to set
      * @param <C>         the type of the sub-context
      */
-    public <C extends SubContext> void set(Class<C> contextType, C instance) {
+    public <C extends SubContext> void set(@Nonnull Class<C> contextType, @Nonnull C instance) {
         subContexts.put(contextType, instance);
     }
 
+    /**
+     * Returns the sub context of the given type.
+     * <p>
+     * Note: In contrast to {@link #get(Class)}, this method will not create a new instance if none is present.
+     *
+     * @param contextType the type of the sub-context to be returned
+     * @param <C>         the type of the sub-context
+     * @return an instance of the given type or <tt>null</tt> if no instance was available
+     */
+    @Nullable
+    @SuppressWarnings("unchecked")
+    public <C extends SubContext> C tryGetSubContext(@Nonnull Class<C> contextType) {
+        return (C) subContexts.get(contextType);
+    }
```

Comment:
The function name originally made me believe the return type would be an `Optional` 😅

Reply:
Yes, I prefer if an Optional comes out if the method is called `try***`
Otherwise rename the other one to getRequredSubContext or the like...